hive/src/main/java/org/apache/iceberg/hive/CustomHiveTableOperations.java

```diff
@@ -0,0 +1,253 @@
+import org.apache.hadoop.hive.metastore.api.NoSuchObjectException;
+import org.apache.hadoop.hive.metastore.api.SerDeInfo;
+import org.apache.hadoop.hive.metastore.api.StorageDescriptor;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.exceptions.CommitFailedException;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.thrift.TException;
+
+
+/**
+ * A {@link HiveTableOperations} that does not override any existing Hive metadata.
+ *
+ * The behaviour of this class differs from {@link HiveTableOperations} in the following ways:
+ * 1. Does not modify serde information of existing Hive table, this means that if Iceberg schema is updated
+ *    Hive schema will remain stale
+ * 2. If the Hive table already exists, no error is thrown. Instead Iceberg metadata is added to the table
+ *
+ * This behaviour is useful if the Iceberg metadata is being generated/updated in response to Hive metadata being
+ * updated.
+ */
+public class CustomHiveTableOperations extends HiveTableOperations {
+  private final HiveClientPool metaClients;
+  private final String database;
+  private final String tableName;
+
+  protected CustomHiveTableOperations(Configuration conf, HiveClientPool metaClients, String database,
+      String table) {
+    super(conf, metaClients, database, table);
+    this.metaClients = metaClients;
+    this.database = database;
+    this.tableName = table;
+  }
+
+  @Override
+  protected void doRefresh() {
+    String metadataLocation = null;
+    try {
+      final Table table = metaClients.run(client -> client.getTable(database, tableName));
+      String tableType = table.getParameters().get(TABLE_TYPE_PROP);
+
+      if (tableType == null || !tableType.equalsIgnoreCase(ICEBERG_TABLE_TYPE_VALUE)) {
+        // if table type is not Iceberg, do not throw an error. Instead just continue, current metadata will still be
+        // null so further operations should work correctly
+      } else {
+        metadataLocation = table.getParameters().get(METADATA_LOCATION_PROP);
+        if (metadataLocation == null) {
+          String errMsg = String.format("%s.%s is missing %s property", database, tableName, METADATA_LOCATION_PROP);
+          throw new IllegalArgumentException(errMsg);
+        }
+      }
+    } catch (NoSuchObjectException e) {
+      if (currentMetadataLocation() != null) {
+        throw new NoSuchTableException(String.format("No such table: %s.%s", database, tableName));
+      }
+
+    } catch (TException e) {
+      String errMsg = String.format("Failed to get table info from metastore %s.%s", database, tableName);
+      throw new RuntimeException(errMsg, e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted during refresh", e);
+    }
+
+    refreshFromMetadataLocation(metadataLocation);
```

Comment:
Seems like `refreshFromMetadataLocation` can be called with a null metadataLocation. Should we move this in the else block?